DeepSpeech.ipynb

```diff
@@ -1163,13 +1183,28 @@
    },
    "outputs": [],
    "source": [
-    "def print_wer_report(session, caption, tower_decodings, tower_labels, show_example=True):\n",
-    "    items, mean = calculate_wer(session, tower_decodings, tower_labels)\n",
-    "    print \"%s WER: %f09\" % (caption, mean)\n",
-    "    if len(items) > 0 and show_example:\n",
-    "        print \"Example (WER = %f09)\" % items[-1][2]\n",
-    "        print \" - source: \\\"%s\\\"\" % items[-1][0]\n",
-    "        print \" - result: \\\"%s\\\"\" % items[-1][1] \n",
+    "def print_wer_report(session, caption, tower_decodings, tower_labels, tower_total_losses, show_ranked=True):\n",
+    "    items, mean = calculate_wer(session, tower_decodings, tower_labels, tower_total_losses)\n",
+    "    print\n",
+    "    print \"#\" * 80\n",
+    "    print \"%s WER: %f\" % (caption, mean)\n",
+    "    if len(items) > 0 and show_ranked:\n",
+    "        # Filter out all items with WER=0\n",
+    "        items = [a for a in items if a[2] > 0]\n",
+    "        # Order the remaining items by their loss (lowest loss on top)\n",
+    "        items.sort(key=lambda a: a[3])\n",
+    "        # Take only the first 10 items\n",
+    "        items = items[:10]\n",
+    "        # Order this top ten items by their WER (lowest WER on top)\n",
+    "        items.sort(key=lambda a: a[2])\n",
+    "        for a in items:\n",
+    "            print \"-\" * 80\n",
+    "            print \" - WER:    %f\" % a[2]\n",
+    "            print \" - source: \\\"%s\\\"\" % a[0]\n",
+    "            print \" - result: \\\"%s\\\"\" % a[1] \n",
+    "            print \" - loss:   %f\" % a[3]\n",
+    "    print \"#\" * 80\n",
+    "    print\n",
     "    return items, mean"
```

Comment:
Improved this now by splitting the method up into a calculation and printing part - which is also much cleaner from a design point of view.